app/services/testing_ground/tree_sampler.rb

```diff
@@ -0,0 +1,27 @@
+class TestingGround
+  class TreeSampler
+    def initialize(networks)
+      @networks = networks
+    end
+
+    def sample(nodes = [], resolution)
+      Hash[@networks.each_pair.map do |carrier, graph|
+        graph.nodes.each do |node|
+          node.set(:load, downsample(node.load, resolution))
+        end
+
+        [carrier, GraphToTree.convert(graph)]
+      end]
+    end
+
+    private
+
+    def downsample(node_load, resolution)
+      if resolution == :low
+        node_load.each_slice(96).map(&:max)
```

Comment:
Almost. The maximum *deviation* from zero should be selected. If you have a slice of two values [10, -15], then **-15** would be the one chosen.